cro-clib/chain.h

```diff
@@ -0,0 +1,44 @@
+/*
+chain c bindings
+*/
+
+#define MAX_LENGTH 1024
+#define MAX_STRING_LENGTH 512
+/* general byte buffer */
+typedef struct {
+    unsigned char buf[MAX_STRING_LENGTH];
+    int length;
+} Buf;
+
+/* general api result */
+typedef struct {
+    int error;
+    Buf value;
+} ApiResult;
+
+/* hdwallet info */
+typedef struct {
+    int error;
+    Buf name;
+    Buf value;
+    Buf mnemonics;
+    Buf viewkey;
+    Buf seed;
+    Buf enckey;
+} HDWallet;
+
+/* initialize state */
+typedef struct {
+    int error;
+    Buf chain_id;
+    Buf server_url;
+    Buf storage_folder;
+} ApiContext;
+
+extern ApiContext initialize(const char* chain_id, const char* server, const char* storage);
```

Comment:
```suggestion
extern "C" ApiContext initialize(const char* chain_id, const char* server, const char* storage);
```

Reply:
this is auto-matically generated by bindgen, but i'll check